### ISSUE TITLE
Run Danger-prose over diffs, not modified files?

### ISSUE THREAD

**Issue by suchow** [dbgrandi/danger-proselint#13]
It seems that this plugin runs over the entire modified file, including text that was merged in previous commits but which is not modified in the PR under consideration.  See, for example, https://github.com/amperser/proselint/pull/609#issuecomment-243230415. Though these suggestions are valid, they are not not relevant to the PR and thus should not be included.

Fixing this issue is complicated because it requires determining which errors raised can be attributed to the PR. One possible solution is to run the prose linters twice, once over the proposed text and once over the original text. Any error that is flagged in the former that did not appear in the latter can be attributed to the proposed change. A second possible solution is to attribute to the PR any error that appears on a modified line. This will falsely flag existing errors that are adjacent to changes, but is a step in the right direction.

**Comment by orta** [comment]
I have an example of doing this kind of analysis in the [Dangerfile for Danger.Systems](https://github.com/danger/danger.systems/blob/master/Dangerfile)

``` ruby
diff = git.diff_for_file("plugins.json")
if diff
  current_plugins = JSON.parse(diff.blob(:dst).contents)
  pr_plugins = JSON.parse(diff.blob(:src).contents)
```

Where plugins.json may have changed from `["thing"]` to `["thing", "other"]` and then I can pull out "other". This could be used to get the before and after, and do the analysis on that. 

I'm not sure how that handles new files though, but I think this is a good idea overall 👍
